src/Akeneo/Pim/Automation/DataQualityInsights/back/Infrastructure/Subscriber/AttributeGroup/UpdateAttributeGroupActivationSubscriber.php

```diff
@@ -0,0 +1,110 @@
+    private $logger;
+
+    public function __construct(
+        FeatureFlag $dataQualityInsightsFeature,
+        AttributeGroupActivationRepositoryInterface $attributeGroupActivationRepository,
+        GetAttributeGroupActivationQueryInterface $getAttributeGroupActivationQuery,
+        LoggerInterface $logger
+    ) {
+        $this->dataQualityInsightsFeature = $dataQualityInsightsFeature;
+        $this->attributeGroupActivationRepository = $attributeGroupActivationRepository;
+        $this->getAttributeGroupActivationQuery = $getAttributeGroupActivationQuery;
+        $this->logger = $logger;
+    }
+
+    public static function getSubscribedEvents()
+    {
+        return [
+            StorageEvents::POST_SAVE => ['createAttributeGroupActivation'],
+            StorageEvents::POST_REMOVE => ['removeAttributeGroupActivation'],
+        ];
+    }
+
+    public function createAttributeGroupActivation(GenericEvent $event): void
+    {
+        $attributeGroup = $event->getSubject();
+        if (! $attributeGroup instanceof AttributeGroupInterface) {
+            return;
+        }
+
+        if (! $this->dataQualityInsightsFeature->isEnabled()) {
+            return;
+        }
+
+        try {
+            $attributeGroupCode = new AttributeGroupCode($attributeGroup->getCode());
+            $attributeGroupActivation = $this->getAttributeGroupActivationQuery->byCode($attributeGroupCode);
+
+            if (null === $attributeGroupActivation) {
+                $attributeGroupActivation = new AttributeGroupActivation($attributeGroupCode, true);
+                $this->attributeGroupActivationRepository->save($attributeGroupActivation);
+            }
+        } catch (\Throwable $exception) {
+            $this->logger->error(
+                sprintf('Unable to create attribute group activation for "%s"', $attributeGroup->getCode()),
+                [
+                    'error_code' => 'unable_to_create_attribute_group_activation',
+                    'error_message' => $exception->getMessage(),
+                ]
+            );
+        }
+    }
+
+    public function removeAttributeGroupActivation(GenericEvent $event): void
+    {
+        $attributeGroup = $event->getSubject();
+        if (! $attributeGroup instanceof AttributeGroupInterface) {
+            return;
+        }
+
+        if (! $this->dataQualityInsightsFeature->isEnabled()) {
+            return;
+        }
+
+        try {
+            $attributeGroupCode = new AttributeGroupCode($attributeGroup->getCode());
+            $this->attributeGroupActivationRepository->remove($attributeGroupCode);
+        } catch (\Throwable $exception) {
+            $this->logger->error(
+                sprintf('Unable to remove attribute group activation for "%s"', $attributeGroup->getCode()),
+                [
+                    'error_code' => 'unable_to_remove_attribute_group_activation',
+                    'error_message' => $exception->getMessage(),
+                ]
```

Comment:
Let the logger manage this, better for structured log exploitation !

```suggestion
                'Unable to remove attribute group activation for "{attributegroup_code}"',
                [
                    'attributegroup_code' => $attributeGroup->getCode(),
                    'error_code' => 'unable_to_remove_attribute_group_activation',
                    'error_message' => $exception->getMessage(),
                ]
```